src/DssDirectDepositAaveDai.sol

```diff
@@ -246,94 +282,144 @@ contract DssDirectDepositAaveDai {
         emit Wind(amount);
     }
 
-    function unwind(uint256 amount, uint256 fees) external {
-        require(wards[msg.sender] == 1 || live == 0, "DssDirectDepositAaveDai/not-authorized");
+    function _unwind(uint256 supplyReduction, uint256 availableLiquidity, uint256 mode) internal {
+        bytes32 ilk_ = ilk;
+        address vow = chainlog.getAddress("MCD_VOW");
+        address end;
+        uint256 adaiBalance = adai.balanceOf(address(this));
+
+        uint256 daiDebt;
+        if (mode == 1) {
```

Comment:
Suggestion - would be nice to use enums for these modes.
For example instead of mode "1"/"2"/.. we will have something like:
Mode.ONLY_CAGED
Mode.CULLED
Mode.MCD_CAGED
Mode.CULLED_THEN_MCD_CAGED

Reply:
Modified to use enum.